pkg/sql/pg_catalog.go

```diff
@@ -378,19 +378,37 @@ CREATE TABLE pg_catalog.pg_attribute (
 			// addColumn adds adds either a table or a index column to the pg_attribute table.
 			addColumn := func(column *sqlbase.ColumnDescriptor, attRelID tree.Datum, colID sqlbase.ColumnID) error {
 				colTyp := &column.Type
+				attTypMod := -1
+				width := int(colTyp.Width())
```

Comment:
nit: if this isn't used outside the if, might make more sense to `if width := ...; width != 0`